api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseEncounterDao.java

```diff
@@ -101,6 +102,7 @@ protected void handleHasAndListParam(Criteria criteria, HasAndListParam hasAndLi
 							// Constrain only on non-voided Drug Orders
 							criteria.add(Restrictions.eq("orders.class", DrugOrder.class));
 							criteria.add(Restrictions.eq("orders.voided", false));
+							criteria.add(Restrictions.ne("orders.action", Order.Action.DISCONTINUE));
```

Comment:
this makes sure the Encounter "has" MedicationRequests functionality doesn't pull back encounters than only have "discontinue" orders.